source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -47,6 +47,9 @@ namespace TcpProxy {
   COUNTER(downstream_flow_control_resumed_reading_total)                                           \
   COUNTER(idle_timeout)                                                                            \
   COUNTER(max_downstream_connection_duration)                                                      \
+  COUNTER(on_demand_cluster_attempt)                                                               \
```

Comment:
Should there be a stat for how many attempts have completed successfully?

Reply:
That will be a low signal counter. The expectation is that a cluster is requested and it's always there until TTL. 

I am happy to add though

Reply:
added